File: src/main/java/com/box/BoxTester.java
package com.box;

import java.util.Scanner;

public class BoxTester {

    public static void main(String[] args) {
        float length;
        float width;
        int height;

        Box3 box3 = new Box3();
        Box5 box5 = new Box5();

        Scanner scanner = new Scanner(System.in);
        System.out.println("請輸入寄送物品\"長度\"");
        length = scanner.nextFloat();

        System.out.println("請輸入寄送物品\"寬度\"");
        width = scanner.nextFloat();

        System.out.println("請輸入寄送物品\"高度\"");
        height = scanner.nextInt();

        if (box5.validate(length,width,height)){
            if (box3.validate(length,width,height)) {
                System.out.println("可放入Box3盒子");
            } else {
                System.out.println("可放入Box5盒子");
            }
        } else {
            System.out.println("沒有合適的盒子!");
        }
    }
}

File: src/main/kotlin/com/box/BoxTester.kt
package com.box.kotlin

import java.util.*

fun main(args: Array<String>) {
    var box3 = Box3(23f,24f,13)
    var box5 = Box5(39.5f,27.5f,23)
    //box3.print();
    val scanner = Scanner(System.`in`);

        println("請輸入寄送物品\"長度\"");
    val length = scanner.nextFloat();

        println("請輸入寄送物品\"寬度\"");
    val width = scanner.nextFloat();

        println("請輸入寄送物品\"高度\"");
    val height = scanner.nextInt();

    if (box5.validate(length, width, height))
        if (box3.validate(length, width, height)) {
            println("可放入Box3盒子");
        } else {
            println("可放入Box5盒子");
        }
    else {
        println("沒有合適的盒子!");
    }
}

class Box3(_length:Float, _width:Float, _height:Int) : Box(_length, _width, _height) {
    override fun validate(length:Float, width:Float, height:Int) : Boolean =
        (_length >= length && _width >= width && _height >= height)
}

class Box5(_length:Float, _width:Float, _height:Int) : Box(_length, _width, _height) {
    override fun validate(length:Float, width:Float, height:Int): Boolean =
        (_length >= length && _width >= width && _height >= height)
}



open class Box(var _length:Float,  var _width:Float,  var _height:Int) {
    open fun validate(length:Float, width:Float, height:Int): Boolean = true
}



